src/clojure/puppetlabs/ssl_utils/core.clj

```diff
@@ -561,12 +561,17 @@
     (SSLUtils/pemToCRLs r)))
 
 (schema/defn ^:always-validate pem->ca-crl :- X509CRL
-  "Given a CRL chain, extract the first CRL"
-  [crl-chain :- Readerable]
-  (let [[crl rest-of-chain] (pem->crls crl-chain)]
+  "Given a CRL chain and CA certificate, extract the CRL issued by the
+  certificate"
+  [crl-chain :- Readerable
+   ca-cert :- X509Certificate]
+  (let [match-ca-cert? (fn [crl] (= (.getIssuerX500Principal crl)
+                                    (.getSubjectX500Principal ca-cert)))
```

Comment:
there any value in saving this to variable instead of calling getSubjectX500Principal every time?